lib/cstruct.ml

```diff
@@ -314,7 +330,10 @@ let iter lenfn pfn buf =
           body := None;
           None
         |Some plen ->
-          let p,rest = split buf plen in
+          incr i;
+          let p,rest =
+            try split buf plen with Invalid_argument _ -> err_iter !i plen
```

Comment:
Not directly, but the `Some plen` value is returned by the `lenfn` function provided by the user (which is called multiple times on the buffer). I think it is useful to know how many time that function has been called by `iter` (the `i`) and what splitting it is trying to do (the `plen`).